tools/solidityUpgrade/SourceTransform.h

```diff
@@ -102,11 +125,22 @@ class SourceTransform
 		std::string const& _expression
 	)
 	{
-		return regex_replace(
-			_location.text(),
-			std::regex{"(\\b" + _keyword + "\\b)"},
-			_expression + " " + _keyword
-		);
+		auto _regex = std::regex{"(\\b" + _keyword + "\\b)"};
+		if (regex_search(_location.text(), _regex))
+			return regex_replace(
+				_location.text(),
+				_regex,
+				_expression + " " + _keyword
+			);
+		else
+			solAssert(
+				false,
```

Comment:
Isn't it too harsh to assert false here? Could instead just report a warning

Reply:
I don't think that a warning would be enough, because the program has to be halted at this point. Otherwise, the same context would be picked again by `solidity-upgrade`, the regex match will fail; this keeps going to an infinite loop.

Reply:
Ah ok!

Reply:
A different option is to implement a `langutil::ErrorReporter` member probably inside `SourceUpgrade`, keep checking for errors before recompiling.